File: frontend/src/api/career-analysis.ts
import { request } from "./client";

/* ── 型定義 ──────────────────────────────────────────── */

export type EvidenceSource = "resume" | "github" | "blog" | "basic_info";
export type Horizon = "short" | "mid" | "long";

export interface TechStackItem {
  name: string;
  priority: 1 | 2 | 3;
  source: string;
  note?: string;
}

export interface StrengthItem {
  title: string;
  detail: string;
  evidence_source: EvidenceSource;
}

export interface CareerPathItem {
  horizon: Horizon;
  label: string;
  title: string;
  description: string;
  required_skills: string[];
  gap_skills: string[];
  fit_score: number;
}

export interface ActionItem {
  priority: number;
  action: string;
  reason: string;
}

export interface CareerAnalysisResult {
  growth_summary: string;
  tech_stack: { top: TechStackItem[]; summary: string };
  strengths: StrengthItem[];
  career_paths: CareerPathItem[];
  action_items: ActionItem[];
}

export interface CareerAnalysisResponse {
  id: number;
  version: number;
  target_position: string;
  result: CareerAnalysisResult | null;
  status: string;
  error_message?: string;
  created_at: string;
}

export interface TaskStatusResponse {
  status: string;
  error_message?: string;
  error_code?: string;
  error_id?: string;
  retry_after?: number;
}

/* ── API 関数 ─────────────────────────────────────────── */

/** キャリアパス分析を開始する（202 非同期）。 */
export function generateAnalysis(targetPosition: string): Promise<CareerAnalysisResponse> {
  return request<CareerAnalysisResponse>("/api/career-analysis/generate", {
    method: "POST",
    body: JSON.stringify({ target_position: targetPosition }),
  });
}

/** 全分析結果を取得する。 */
export function listAnalyses(): Promise<CareerAnalysisResponse[]> {
  return request<CareerAnalysisResponse[]>("/api/career-analysis/");
}

/** 指定 ID の分析結果を取得する。 */
export function getAnalysis(id: number): Promise<CareerAnalysisResponse> {
  return request<CareerAnalysisResponse>(`/api/career-analysis/${id}`);
}

/** 分析ステータスを取得する（ポーリング用）。 */
export function getAnalysisStatus(id: number): Promise<TaskStatusResponse> {
  return request<TaskStatusResponse>(`/api/career-analysis/${id}/status`);
}

/** 分析結果を削除する。 */
export function deleteAnalysis(id: number): Promise<void> {
  return request<void>(`/api/career-analysis/${id}`, {
    method: "DELETE",
  });
}

File: backend/app/schemas/career_analysis.py
"""AI キャリアパス分析のリクエスト/レスポンススキーマ。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TechStackItem(BaseModel):
    """技術スタック1件。"""

    name: str
    priority: int  # 1〜3
    source: str  # "案件実績" | "個人開発" | "資格"
    note: str | None = None


class TechStackSection(BaseModel):
    """技術スタック評価セクション。"""

    top: list[TechStackItem]
    summary: str


class StrengthItem(BaseModel):
    """強み1件（根拠付き）。"""

    title: str
    detail: str
    evidence_source: str  # "resume" | "github" | "blog" | "basic_info"


class CareerPathItem(BaseModel):
    """キャリアパス提案1件。"""

    horizon: str  # "short" | "mid" | "long"
    label: str
    title: str
    description: str
    required_skills: list[str]
    gap_skills: list[str]
    fit_score: int  # 0〜100


class ActionItem(BaseModel):
    """アクションアイテム1件。"""

    priority: int
    action: str
    reason: str


class CareerAnalysisResult(BaseModel):
    """キャリアパス分析の全セクション。"""

    growth_summary: str
    tech_stack: TechStackSection
    strengths: list[StrengthItem]
    career_paths: list[CareerPathItem]  # short/mid/long の 3件
    action_items: list[ActionItem]


class CareerAnalysisGenerateRequest(BaseModel):
    """キャリアパス分析の生成リクエスト。"""

    target_position: str = Field(max_length=200)


class CareerAnalysisResponse(BaseModel):
    """キャリアパス分析のレスポンス。"""

    id: int
    version: int
    target_position: str
    result: CareerAnalysisResult | None = None
    status: str = "completed"
    error_message: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TaskStatusResponse(BaseModel):
    """タスクステータスの軽量レスポンス。"""

    status: str
    error_message: str | None = None
    error_code: str | None = None
